src/index.js

```diff
@@ -366,32 +328,50 @@ class IPLDResolver {
         } else { // multicodec unknown, send back raw data
           cb(null, block.data)
         }
-      })
-    )
+      }
+    ], callback)
   }
 
-  _putStream (callback) {
+  _put (cid, node, callback) {
     callback = callback || noop
 
-    return pull(
-      pull.asyncMap((nodeAndCID, cb) => {
-        const cid = nodeAndCID.cid
-        const r = this.resolvers[cid.codec]
-
-        r.util.serialize(nodeAndCID.node, (err, serialized) => {
-          if (err) {
-            return cb(err)
-          }
-          cb(null, {
-            block: new Block(serialized),
-            cid: cid
-          })
-        })
-      }),
-      this.bs.putStream(),
-      pull.onEnd(callback)
-    )
+    const r = this.resolvers[cid.codec]
+    waterfall([
+      (cb) => r.util.serialize(node, cb),
+      (buf, cb) => this.bs.put(new Block(buf, cid), cb)
+    ], (err) => {
+      if (err) {
+        return callback(err)
+      }
+      callback(null, cid)
+    })
   }
 }
 
+/**
+ * Create an IPLD resolver with an inmemory blockservice and
+ * repo.
+ *
+ * @param {function(Error, IPLDResolver)} callback
+ * @returns {void}
+ */
+IPLDResolver.inMemory = function (callback) {
+  const repo = new IPFSRepo('in-memory', {
+    fs: MemoryStore,
+    level: require('memdown'),
+    lock: 'memory'
+  })
+  const blockService = new BlockService(repo)
+
+  series([
+    (cb) => repo.init({}, cb),
+    (cb) => repo.open(cb)
+  ], (err) => {
+    if (err) {
+      return callback(err)
+    }
+    callback(null, new IPLDResolver(blockService))
+  })
+}
```

Comment:
this turned out to be great, as I had forgotten to test in-memory in nodejs and found a bug in the repo